### PR TITLE
attempt to fix #16374

### LINKED ISSUE #16374: Codegen bug with Table mvalues in loop and ARC & ORC GC
Small example that uses Table.mvalues within a loop fails to compile the generated C with `--gc:arc` and `--gc:orc` flags

Only an issue with `--gc:arc` and `--gc:orc`. All others (`markAndSweep`, `boehm`, `go`, `none`, `regions`, `refc`) work fine.

Seems like the Table value type being a `seq` is important - it compiles if I change it to an `int`, or even an `array[0..1, int]`.

### Example
```nim
import tables

var t: Table[int, seq[int]]

while false:
  for v in t.mvalues:
    discard
```

### Current Output
With nim 1.4.2
```
 > nim compile --gc:arc src/day16_bug.nim
Hint: used config file '/home/lordaro/dev/nim-1.4.2/config/nim.cfg' [Conf]
Hint: used config file '/home/lordaro/dev/nim-1.4.2/config/config.nims' [Conf]
..........
Hint: gcc -c  -w -fmax-errors=3   -I/home/lordaro/dev/nim-1.4.2/lib -I/home/lordaro/dev/AdventOfCode/2020/src -o /home/lordaro/.cache/nim/day16_bug_d/stdlib_tables.nim.c.o /home/lordaro/.cache/nim/day16_bug_d/stdlib_tables.nim.c [Exec]
Hint: gcc -c  -w -fmax-errors=3   -I/home/lordaro/dev/nim-1.4.2/lib -I/home/lordaro/dev/AdventOfCode/2020/src -o /home/lordaro/.cache/nim/day16_bug_d/@mday16_bug.nim.c.o /home/lordaro/.cache/nim/day16_bug_d/@mday16_bug.nim.c [Exec]
/home/lordaro/.cache/nim/day16_bug_d/@mday16_bug.nim.c: In function ‘NimMainModule’:
/home/lordaro/.cache/nim/day16_bug_d/@mday16_bug.nim.c:225:34: error: ‘value__TeitEu5dFeGB2q0HIvuFvA’ is a pointer; did you mean to use ‘->’?
     value__TeitEu5dFeGB2q0HIvuFvA.len = 0; value__TeitEu5dFeGB2q0HIvuFvA.p = NIM_NIL;
                                  ^
                                  ->
/home/lordaro/.cache/nim/day16_bug_d/@mday16_bug.nim.c:225:73: error: ‘value__TeitEu5dFeGB2q0HIvuFvA’ is a pointer; did you mean to use ‘->’?
     value__TeitEu5dFeGB2q0HIvuFvA.len = 0; value__TeitEu5dFeGB2q0HIvuFvA.p = NIM_NIL;
                                                                         ^
                                                                         ->
Error: execution of an external program failed: 'gcc -c  -w -fmax-errors=3   -I/home/lordaro/dev/nim-1.4.2/lib -I/home/lordaro/dev/AdventOfCode/2020/src -o /home/lordaro/.cache/nim/day16_bug_d/@mday16_bug.nim.c.o /home/lordaro/.cache/nim/day16_bug_d/@mday16_bug.nim.c'
```

### Expected Output
Successful compile

### Additional Information

```
 > nim -v
Nim Compiler Version 1.4.2 [Linux: amd64]
Compiled at 2020-11-30
Copyright (c) 2006-2020 by Andreas Rumpf

git hash: 3fb5157ab1b666a5a5c34efde0f357a82d433d04
active boot switches: -d:release
```

### PR DESCRIPTION
**Pull request by ringabout**
fix #16374

### CODE REVIEW

File: compiler/cgen.nim [hunk 1]
```diff
@@ -406,7 +406,12 @@ proc resetLoc(p: BProc, loc: var TLoc) =
   if isImportedCppType(typ): return
   if optSeqDestructors in p.config.globalOptions and typ.kind in {tyString, tySequence}:
     assert rdLoc(loc) != nil
-    linefmt(p, cpsStmts, "$1.len = 0; $1.p = NIM_NIL;$n", [rdLoc(loc)])
+
+    let atyp = skipTypes(loc.t, abstractInst)
+    if atyp.kind in {tyVar}:
```
Comment (timotheecour) [inline compiler/cgen.nim]: is `tyLent` also needed ? (I don't know, but the test should also check for lent to make sure it works)

Comment (ringabout) [reply compiler/cgen.nim]: Ok, will add it after seeing whether CI is green.

Comment (ringabout) [reply compiler/cgen.nim]: It doesn't seem to be needed.

This compiles before.
```nim
iterator mvalues(t: var seq[seq[int]]): lent seq[int] =
  yield t[0]

var t: seq[seq[int]]

while false:
  for v in t.mvalues:
    discard

proc ok =
  while false:
    for v in t.mvalues:
      discard

ok()

while false:
  for v in t.mvalues:
    discard
```

Comment (Araq) [reply compiler/cgen.nim]: It's safer to include the `tyLent` though.

Comment (ringabout) [reply compiler/cgen.nim]: will do after CI is green.

**Pull request merged**